test/test_function_data.py

```diff
@@ -267,5 +267,185 @@ def test_export_h5(self):
+            ]
+        }
+        target_solution_data = ForwardSolutionData(
+            target_time_partition, target_function_spaces
+        )
+        target_solution_data._create_data()
+        with self.assertRaises(ValueError) as cm:
+            self.solution_data.transfer(target_solution_data, method="interpolate")
+        self.assertEqual(
+            str(cm.exception),
+            "Source and target have different numbers of exports per subinterval.",
+        )
+
+    def test_transfer_common_fields_error(self):
+        target_time_partition = TimePartition(
+            self.time_partition.end_time,
+            self.time_partition.num_subintervals,
+            self.time_partition.timesteps,
+            ["different_field"],
+        )
+        target_function_spaces = {
+            "different_field": [
+                FunctionSpace(self.mesh, "DG", 0)
+                for _ in range(target_time_partition.num_subintervals)
+            ]
+        }
+        target_solution_data = ForwardSolutionData(
+            target_time_partition, target_function_spaces
+        )
+        target_solution_data._create_data()
+        with self.assertRaises(ValueError) as cm:
+            self.solution_data.transfer(target_solution_data, method="interpolate")
+        self.assertEqual(
+            str(cm.exception), "No common fields between source and target."
+        )
+
+    def test_transfer_common_labels_error(self):
+        target_solution_data = ForwardSolutionData(
+            self.time_partition, self.function_spaces
+        )
+        target_solution_data._create_data()
+        target_solution_data.labels = ("different_label",)
+        with self.assertRaises(ValueError) as cm:
+            self.solution_data.transfer(target_solution_data, method="interpolate")
+        self.assertEqual(
+            str(cm.exception), "No common labels between source and target."
+        )
+
+    def test_transfer_interpolate(self):
+        target_solution_data = ForwardSolutionData(
+            self.time_partition, self.function_spaces
+        )
+        target_solution_data._create_data()
+        self.solution_data.transfer(target_solution_data, method="interpolate")
+        for field in self.solution_data.time_partition.field_names:
+            for label in self.solution_data.labels:
+                for i in range(self.solution_data.time_partition.num_subintervals):
+                    for j in range(
+                        self.solution_data.time_partition.num_exports_per_subinterval[i]
+                        - 1
+                    ):
+                        source_function = self.solution_data._data[field][label][i][j]
+                        target_function = target_solution_data._data[field][label][i][j]
+                        self.assertTrue(
+                            source_function.dat.data.all()
+                            == target_function.dat.data.all()
+                        )
```

Comment:
@jwallwork23 I forgot to mention. This doesn't really make sense since the data arrays are just 0. Maybe it's enough to just call the function and see that there are no errors raised?

Reply:
As a minimal test you could set the array to be uniformly 1 and check that is mapped over.

Reply:
Thanks :) done in 33102ec5b77a452182f39f01f1e55addb358afe9